src/UI/Art/SpriteArtist/SpriteMapView.java

```diff
@@ -231,41 +196,42 @@ public void render(Graphics g)
     if( drawY < 0 )
       drawY = 0;
 
+    // Get a reference to the current action being built, if one exists.
+    Unit currentActor = mapController.getContemplatedActor();
+    XYCoord actorCoord = mapController.getContemplationCoord();
+    boolean notifyOnAnimEnd = true;
+    if( null != currentActor && null == currentAnimation ) // Draw the currently-acting unit so it's on top of everything.
+    {
+      currentAnimation = new BaseUnitAnimation(drawMultiplier, currentActor, actorCoord);
```

Comment:
This creates a new animation and destroys it every frame - I feel like there's a better way to handle this.

Reply:
Yeah, probably.

Reply:
It appears to be destroying performance. I can barely run the game on my laptop as soon as I create a unit.

Reply:
Wait, seriously?
That doesn't make any sense.
First, it should only activate once you select a unit. Second, I removed the `new` anyhow.

Reply:
OK, this probably wasn't it, given that it's still an issue with the new changes you made. Something changed in `cyanide` though that is creating huge lag once a unit exists.

Reply:
Looks like it was that loop you indicated. Which prompts me to ask - why did you change how the animation index is calculated?

Reply:
I changed it so it could be a static method, so I could use it without daisy-chaining the state info it's based on around. (and also because there's no particularly great reason IMO for holding effectively-meaningless state info)

I'd have thrown the mod logic in the anim index getter function, but that could cause animation timings to be off because we have no enforced animation length.